src/plotSearch/components/reservationIdentifiers/ReservationIdentifiersModal.js

```diff
@@ -0,0 +1,381 @@
+          reservationIdentifier: target.reservation_readable_identifier,
+          intendedUse: (target.plan_unit?.intended_use || target.custom_detailed_plan?.intended_use)?.name,
+          applicants: target.reservation_recipients.map((group) =>
+            `${group.reservation_recipients.join(', ')} (${group.share_of_rental})`).join('; '),
+          leaseOptions,
+          disabled: !!target.reservation_identifier,
+        },
+      ];
+    }, []);
+
+    this.setState(() => ({
+      rowData: data,
+    }));
+    initialize({
+      rowsToCreate: data.map((row) => ({
+        selected: !row.reservationIdentifier,
+      })),
+    });
+  }
+
+  selectEditorTarget = (index: number | null, row: Object) => {
+    const {initializeForm} = this.props;
+
+    if (index !== null) {
+      initializeForm(FormNames.LEASE_CREATE_MODAL, {
+        [LeaseFieldPaths.MUNICIPALITY]: row.leaseOptions.municipality,
+        [LeaseFieldPaths.DISTRICT]: row.leaseOptions.district,
+        [LeaseFieldPaths.TYPE]: row.leaseOptions.type,
+        [LeaseFieldPaths.STATE]: row.leaseOptions.state,
+      });
+    }
+    this.setState(() => ({
+      editorTarget: index,
+    }));
+  };
+
+  handleEditFinished = (leaseData: Object) => {
+    if (this.state.editorTarget !== null) {
+      const index = this.state.editorTarget;
+
+      this.setState((state) => {
+        const newRowData = [...state.rowData];
+        newRowData.splice(index, 1, {
+          ...state.rowData[index],
+          suggestedLeaseIdentifier: this.convertLeaseOptionsToIdentifier(leaseData),
+          leaseOptions: leaseData,
+        });
+
+        return ({
+          rowData: newRowData,
+        });
+      });
+    }
+  };
+
+  handleSubmit = () => {
+    const {
+      batchCreateReservationIdentifiers,
+      onClose,
+      fetchSinglePlotSearch,
+      currentPlotSearch,
+      rowsToCreate,
+    } = this.props;
+    const {rowData} = this.state;
+
+    batchCreateReservationIdentifiers({
+      data: rowData.filter((row) => rowsToCreate[row.index].selected).map((row) => ({
+        ...row.leaseOptions,
+        targetId: row.targetId,
+        leaseId: row.leaseId,
+      })),
+      callback: () => {
+        onClose();
+        fetchSinglePlotSearch(currentPlotSearch.id);
+      },
+    });
+  };
+
+  convertLeaseOptionsToIdentifier = (options: Object) => {
+    const {reservationIdentifierUnitLists} = this.props;
+
+    if (!reservationIdentifierUnitLists) {
+      return '...';
+    }
+
+    const type = reservationIdentifierUnitLists.types.find((candidate) => candidate.id === options.type);
+    const municipality = reservationIdentifierUnitLists.municipalities.find((candidate) => candidate.id === options.municipality);
+    const district = reservationIdentifierUnitLists.districts.find((candidate) => candidate.id === options.district);
+
+    return `${type.identifier || '??'}${municipality.identifier || '?'}${padStart(district.identifier || '??', 2, '0')}`;
```

Comment:
Type can be undefined if backend is failing.

Reply:
Hmm, I thought that shouldn't be possible. The saga that fetches the three lists shouldn't trigger the action to set them into the reducer if any of them couldn't be fetched, but it's possible I made a mistake there somewhere.

Reply:
Right, it wasn't because of that but because the plot search targets were missing the newly added fields. While it probably won't happen with the correct BE, I'll fix the fallbacks to also trigger if any unit was not found in the lists, because that's what they were supposed to do from the outset.